### PR TITLE
T25382-rebase-debian

### PR DESCRIPTION
**Pull request by uajain**
2 commits: 
* Import debian packing files from totem_3.31.91-1.debian.tar.xz 

* Rebase previous Endless-specific changes on top
  (Dropped changes from 2dc50f7 and c51bdae as no longer required) 

Endless specific changes:
  * Add eoscontent integration (commit: b3f8455)
    **keep**

  * Don't recommend gstreamer1.0-plugins-ugly (commit: 2dc50f7)
  **dropped** as [3.31.91 packaging](https://salsa.debian.org/gnome-team/totem/commit/7792762e2fcb00156f6d62cb6634b9fbafb69efd) has taken care of it.

  * Disable Zeitgeist plugin (commit: c51bdae)
     **dropped** as [3.31.91 packaging](https://salsa.debian.org/gnome-team/totem/commit/675d644e51a2c41a2946d7123f0a3a58402d730b) has taken care of it
  * Try to fix Jenkins build (commit: c9d13c0)
    ~~**keep**~~ **dropped**

https://phabricator.endlessm.com/T25382

### CODE REVIEW

File: debian/patches/debian/Don-t-run-pylint-at-build-time.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
+From: Simon McVittie <smcv@debian.org>
+Date: Mon, 2 Jul 2018 18:41:57 +0100
+Subject: Don't run pylint at build time
```
Comment (wjt) [inline debian/patches/debian/Don-t-run-pylint-at-build-time.patch]: If you're still wondering: this patch is the reason that the pylint build failure went away once the source and packaging branches had matching names.

This patch should not be here. It's not made absolutely explicit on https://phabricator.endlessm.com/w/howto/rebase_debian_packages/ but the point of this instruction:

> 5. Inside the `debian/patches/` directory, set aside the patches that we want to keep (they'll be applied later).

is that we do not typically keep patches in `debian/patches/` on the packaging branch; we apply them to our code branch. So the packaging branch should not contain a `debian/patches` directory, and (if we want to keep this patch) it should be applied on the code branch.

Comment (uajain) [reply debian/patches/debian/Don-t-run-pylint-at-build-time.patch]: Oh, okay. Yes, the docs doesn't clarify it.

Hence, I suppose [this](https://github.com/endlessm/totem/pull/8/commits/90f789011fb71719eef9f0a0ca150e06898cafbf) and [this](https://github.com/endlessm/totem/pull/8/commits/6c864644e4e82d333d1e912a18612573e974f8e9) are patches from `3-26` packaging branch that are applied to code branch. And I have rebased them over again. Hence, those needs to drop right? 

I will apply `debian/patches/debian/Don-t-run-pylint-at-build-time.patch` patch to the code branch instead. Thanks for taking a look!

Comment (wjt) [reply debian/patches/debian/Don-t-run-pylint-at-build-time.patch]: Those 2 look like Endless-specific changes, not patches inherited from Debian – why would they need to be dropped? They could probably do with better commit messages. Perhaps you could do some Phabricator archaeology to work out why they were added, and (in the case of the first one) whether they could be upstreamed.

Comment (uajain) [reply debian/patches/debian/Don-t-run-pylint-at-build-time.patch]: heh right. Another **fast** reading fiasco. Just saw the `.patch` extension and assumed that they are applied on the code from previous debian packaging branch.

(My brain is not helping today :/ )

**Comment by uajain** [comment]
retest this please

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]
The last commit should at least be properly worded. "Try to fix Jenkins build" isn't the pinnacle of good commit messages :stuck_out_tongue: 

Other than that, looks good.

**Comment by uajain** [comment]
> The last commit should at least be properly worded. "Try to fix Jenkins build" isn't the pinnacle of good commit messages 
> 
> Other than that, looks good.

Summoning @cosimoc here :stuck_out_tongue: 
As I have no idea about that specific dependency 
@cosimoc Do you still remember why we needed `libgl1-mesa-dri` in  c9d13c036c5 ?

**Comment by cosimoc** [comment]
@uajain The relevant ticket was https://phabricator.endlessm.com/T18853 I think. I can't remember all the details, but IIRC the gtk-doc scanner would not run successfully in Jenkins unless that mesa package was installed. It may be that it's not required anymore, not sure...

**Comment by wjt** [comment]
Personally I think the two commits squashed into "Rebase previous changes on top" should be two separate commits, and that dropping the "debian/patches" should be its own commit.

**Comment by uajain** [comment]
> Personally I think the two commits squashed into "Rebase previous changes on top" should be two separate commits, 

hmm, probably. But this one works too as "Rebase previous changes on top" acts a reference/index commit to all the endless-specific commits. Also, I just followed how it was [done earlier](https://github.com/endlessm/totem/commit/5944fb479477b5fc7435cb60fadf278f58873b5e) :stuck_out_tongue:  and I found it quite logical. 

>and that dropping the "debian/patches" should be its own commit.

For this one, I agree, perhaps we can make it as a "convention". 
* Pristine debian import -> drop debian /patches -> Endless-specific commits

We can mention this convention to be followed in the wiki document.

**Comment by uajain** [comment]
I dropped "Try to fix Jenkins build (commit: c9d13c0)" in my local testing and the branch seems to build successfully via jenkins script. I will now push a `fixup!` commit and retest on jenkins again.

**Comment by uajain** [comment]
> I dropped "Try to fix Jenkins build (commit: [c9d13c0](https://github.com/endlessm/totem/commit/c9d13c036c5878ae44665644aa5e037b77b2023f))" in my local testing and the branch seems to build successfully via jenkins script. I will now push a `fixup!` commit and retest on jenkins again.

Yup. It worked. Dropping "Try to fix Jenkins build (commit: c9d13c0)"

**Review by GeorgesStavracas — APPROVED** [review]
Looks good to me now, although [Totem 3.32.0-1 is apparently already available](https://salsa.debian.org/gnome-team/totem).

**Pull request merged**